plugin.php

```diff
@@ -17,36 +17,159 @@
 
 namespace Alley\WP\REST_API_Guard;
 
+use WP_Error;
+use WP_REST_Request;
+use WP_REST_Server;
+
 if ( ! defined( 'ABSPATH' ) ) {
 	exit;
 }
 
-// Check if Composer is installed.
-if ( ! file_exists( __DIR__ . '/vendor/autoload.php' ) ) {
-	\add_action(
-		'admin_notices',
-		function() {
-			?>
-			<div class="notice notice-error">
-				<p><?php esc_html_e( 'Composer is not installed and rest-api-guard cannot load. Try using a `*-built` branch if the plugin is being loaded as a submodule.', 'plugin_domain' ); ?></p>
-			</div>
-			<?php
-		}
-	);
+/**
+ * Instantiate the plugin.
+ */
+function main() {
+	require_once __DIR__ . '/settings.php';
 
-	return;
+	add_action( 'rest_pre_dispatch', __NAMESPACE__ . '\on_rest_pre_dispatch', 10, 3 );
 }
+main();
+
+/**
+ * Check if anonymous access should be prevented for the current request.
+ *
+ * @param WP_REST_Server  $server  Server instance.
+ * @param WP_REST_Request $request The request object.
+ * @return bool
+ */
+function should_prevent_anonymous_access( WP_REST_Server $server, WP_REST_Request $request ): bool {
+	$settings = (array) get_option( SETTINGS_KEY );
+
+	if ( ! is_array( $settings ) ) {
+		$settings = [];
+	}
+
+	/**
+	 * Check if anonymous access is prevent by default.
+	 *
+	 * @param bool             $prevent Whether to prevent anonymous access, default false.
+	 * @param \WP_REST_Request $request REST API Request.
+	 */
+	if ( true === apply_filters( 'rest_api_guard_prevent_anonymous_access', $settings['prevent_anonymous_access'] ?? false, $request ) ) {
+		return true;
+	}
+
+	$endpoint = $request->get_route();
 
-// Load Composer dependencies.
-require_once __DIR__ . '/vendor/autoload.php';
+	/**
+	 * Prevent access to the root of the REST API.
+	 *
+	 * @param bool $prevent Whether to prevent anonymous access, default false.
+	 */
+	if ( '/' === $endpoint && false === apply_filters( 'rest_api_guard_allow_index_access', $settings['allow_index_access'] ?? false ) ) {
+		return true;
+	}
 
-// Load the plugin's main files.
-require_once __DIR__ . '/src/meta.php';
+	/**
```

Comment:
Is this comment "close enough" to the `apply_filters` call that it gets picked up as being associated with it?